src/variant_detection/variant_detection.cpp

```diff
@@ -60,6 +68,26 @@ void detect_variants_in_alignment_file(const std::filesystem::path & alignment_f
             continue;
 
         const std::string ref_name = ref_ids[ref_id];
+        uint32_t read_length{get_read_length(cigar)};
+        // If there are positions to be sampled, then sample them.
+        if (!rand_sample.empty() && sample_size > 0)
+        {
+            if (ref_id == std::get<0>(rand_sample.front()) &&           // Correct chromosome.
```

Comment:
```suggestion
            auto && [chr, pod] = rand_sample.front();
            if (ref_id == chr) &&           // Correct chromosome.
```

give the pair content a name that makes it easier to understand.